src/error.rs

```diff
@@ -152,6 +152,14 @@ impl std::fmt::Display for UnexpectedMessageType {
 	}
 }
 
+/// The received message had an unexpected service ID.
+#[derive(Debug, Clone, Error)]
+#[error("unexpected service ID: {actual_service_id}")]
+pub struct UnexpectedServiceId {
+	/// The actual service ID of the received message.
+	pub actual_service_id: i32,
```

Comment:
What is the other `service_id` then?

Reply:
The expected service ID, but it might be one value or a whole list of possible values. I don't want to put a vector in the error, so I just left it off (it's not super important anyway). But maybe I should just rename this to `service_id`.

Reply:
Processed on the branch  for generating server structs.